code/modules/client/preferences/preferences_savefile.dm

```diff
@@ -0,0 +1,609 @@
+	if(check.Execute(db))
+		if(!check.NextRow())
+			var/list/arguments_query = new_db_entry_query_args(ckey)
+			q.Add(arglist(arguments_query))
+			if(!q.Execute(db))
+				message_admins("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #: [q.Error()] - [q.ErrorMsg()]. sql= [arguments_query[1]]")
+				WARNING("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #:[q.Error()] - [q.ErrorMsg()]")
+				return 0
+		else
+			var/list/arguments_query = save_db_entry_query_args(ckey)
+			q.Add(arglist(arguments_query))
+			if(!q.Execute(db))
+				message_admins("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #: [q.Error()] - [q.ErrorMsg()]. sql= [arguments_query[1]]")
+				WARNING("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #:[q.Error()] - [q.ErrorMsg()]")
+				return 0
+	else
+		message_admins("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #: [check.Error()] - [check.ErrorMsg()]")
+		WARNING("Error in save_preferences_sqlite [__FILE__] ln:[__LINE__] #:[q.Error()] - [q.ErrorMsg()]")
+		return 0
+	to_chat(user, "Preferences Updated.")
+	lastPolled = world.timeofday
+	return 1
+
+// -- You should not need to modify this - this automatically reads all settings and save them.
+/datum/preferences/proc/new_db_entry_query_args(var/ckey)
+	var/list/returned_list = list()
+	var/sql_text = "INSERT into client (ckey"
+	var/sql_text_end = ") VALUES (?"
+	returned_list.Add(sql_text) // First item in the query is the SQL
+	returned_list.Add(ckey) // Second item is the ckey (first joker)
+	// This is a prepared queries. All the "?" here are jokers which will be replaced internally by BYOND with the values in the param list
+	// This MEANS that the order of the param list is pretty important!
+	for (var/key, setting in preference_settings_client)
+		var/datum/preference_setting/the_setting = setting
+		sql_text += ", [the_setting.sql_name]"
+		sql_text_end += ", ?"
+		returned_list.Add(the_setting.setting)
+	sql_text_end += ")"
+	returned_list[1] = "[sql_text][sql_text_end]"
+	return returned_list
+
+/datum/preferences/proc/save_db_entry_query_args(var/ckey)
+	var/list/returned_list = list()
+	var/sql_text = "UPDATE client SET "
+	var/sql_text_end = " WHERE ckey = ?"
+	returned_list.Add(sql_text) // First item in the query is the SQL
+	// This is a prepared queries. All the "?" here are jokers which will be replaced internally by BYOND with the values in the param list
+	// This MEANS that the order of the param list is pretty important!
+	for (var/key, setting in preference_settings_client)
+		var/datum/preference_setting/the_setting = setting
+		sql_text += "[the_setting.sql_name]=?, "
+		returned_list.Add(the_setting.setting)
+	sql_text = copytext(sql_text, 1, length(sql_text)-1) // Remove the last ,
+	returned_list.Add(ckey) // This time, the ckey is the last joker
+	returned_list[1] = "[sql_text][sql_text_end]"
+	return returned_list
+
+/datum/preferences/proc/load_character_sqlite(var/ckey, var/user, var/slot)
```

Comment:
I have split the procs to load and save a character becuase they were merged into one before, which was very difficult to untangle.